src/temgymbasic/components.py

```diff
@@ -210,10 +210,16 @@ def gui_wrapper():
 
 class Source(Component):
     def __init__(
-        self, z: float, tilt_yx: Tuple[float, float] = (0., 0.), name: Optional[str] = None
+        self, z: float,
+        wavelength: float,
+        random: bool = False,
```

Comment:
I don't think `random` makes sense on the parent class here. If this is for accumulating random rays to a full image we can find a way to write that specifically.